src/IntegrationUtils.ts

```diff
@@ -59,13 +60,55 @@ const choiceList = [
 
  let childProcessMap : Map<string, child_process.ChildProcess>;
 
- export function startIntegration(context: vscode.Uri): Promise<boolean> {
+ export function startIntegration(...args: any[]): Promise<boolean> {
 	return new Promise <boolean> ( async (resolve, reject) => {
-		const choice : string | undefined = await vscode.window.showQuickPick(choiceList, {
-			placeHolder: 'Select the type of Apache Camel K Integration'
-		});
+		let context : vscode.Uri | undefined = undefined;
+		let inChoice : string | undefined = undefined;
+
+		// with no arguments, try working with the selected file
+		if (!args || args.length === 0) {
+			await getCurrentFileSelectionPath().then((fileUri) => {
```

Comment:
I wouldn't mix await and then / catch. 
`try {
   context = await getCurrentFileSelectionPath();
   inChoice = devModeIntegration;
} catch (error) {
   reject('No arguments provided to start integration function call');
   return;	
}

Reply:
Isn't the whole idea of resolve/reject that you can use then/catch?

Reply:
This one I was able to switch to a try/catch.